app/core/metrics.py

```
@@ -0,0 +1,304 @@
+
+
+def timed_calculation(calculation_type: str):
+    """Decorator to time astrological calculations."""
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            start_time = time.time()
+            try:
+                result = func(*args, **kwargs)
+                duration = time.time() - start_time
+                metrics_manager.record_calculation(calculation_type, duration)
+                return result
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)